### PR TITLE
New sidekiq job to poll for 21-526EZ PDF for a submission

### PR DESCRIPTION
**Pull request by sethdarragile6**
The PR creates a new job `PollForm526Pdf` which looks for the expected supporting document (21-526EZ PDF) being produced for a submission asyncronously.
1. If found, it exits out with `success` status, and writes to the Rails log
2. If not found, it retries (with exponential backoff) for 48 hours
3. If, after 48 hours, the supporting doc is still not found, the job exits out with `pdf_not_found` status

At this point, we are only creating the job. A follow-up story will hook it up to the ancillary document upload flow, and/or otherwise be available to be run via an Argo terminal

## Summary

- *This work is behind a feature toggle (flipper): NO*
- *(Summarize the changes that have been made to the platform)*
A new sidekiq job created, but not yet implemented into existing flows
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?)*
Disability Benefits Experience Team 1 Dbex Trex (rawr 🦖)
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
[#84389](https://app.zenhub.com/workspaces/disability-experience-63dbdb0a401c4400119d3a44/issues/gh/department-of-veterans-affairs/va.gov-team/84389)
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: app/sidekiq/lighthouse/poll_form526_pdf.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    include Sidekiq::Form526JobStatusTracker::JobTracker
+    extend Logging::ThirdPartyTransaction::MethodWrapper
+    extend ActiveSupport::Concern
+
+    attr_accessor :submission_id
+
+    # Sidekiq has built in exponential back-off functionality for retries
+    # A max retry attempt of 15 will result in a run time of ~36 hours
+    # Changed from 15 -> 14 ~ Jan 19, 2023
+    # This change reduces the run-time from ~36 hours to ~24 hours
```
Comment (freeheeling) [inline app/sidekiq/lighthouse/poll_form526_pdf.rb:21]: Comment should either be modified or removed.

**Review by freeheeling — COMMENTED** [review]
Will a spec be created for `PollForm526Pdf`?

File: app/sidekiq/lighthouse/poll_form526_pdf.rb (hunk 1 above)
Comment (sethdarragile6) [reply app/sidekiq/lighthouse/poll_form526_pdf.rb]: thanks. killed it.

**Comment by sethdarragile6** [comment]
> Will a spec be created for `PollForm526Pdf`?

@freeheeling yes! just created one

File: app/sidekiq/lighthouse/poll_form526_pdf.rb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+module Lighthouse
+  class PollForm526Pdf
+    include Sidekiq::Job
+    include Sidekiq::Form526JobStatusTracker::JobTracker
+    extend Logging::ThirdPartyTransaction::MethodWrapper
+    extend ActiveSupport::Concern
+
+    attr_accessor :submission_id
+
+    STATSD_KEY_PREFIX = 'worker.lighthouse.poll_form526_pdf'
```
Comment (freeheeling) [inline app/sidekiq/lighthouse/poll_form526_pdf.rb:18]: Where is `STATSD_KEY_PREFIX` used?

Comment (sethdarragile6) [reply app/sidekiq/lighthouse/poll_form526_pdf.rb:18]: @freeheeling confirmed locally that it's behaving as expected! thanks for the callout

File: spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+             auth_headers_json: auth_headers.to_json,
+             saved_claim_id: saved_claim.id)
+    end
+
+    context 'when all retries are exhausted' do
+      let(:form526_job_status) do
+        create(:form526_job_status, :poll_form526_pdf, form526_submission: submission, job_id: 1)
+      end
+
+      it 'transitions the submission to a failure state' do
```
Comment (freeheeling) [inline spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb:33]: Wouldn't a more accurate test description include something related to 'pdf_not_found' being written to the job status?

Comment (sethdarragile6) [reply spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb]: totally meant to update that and forgot. thanks. updated to reflect reality! @freeheeling

File: spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  let(:user) { FactoryBot.create(:user, :loa3) }
+  let(:auth_headers) do
+    EVSS::DisabilityCompensationAuthHeaders.new(user).add_headers(EVSS::AuthHeaders.new(user).to_h)
+  end
+
+  describe '.perform_async' do
+    let(:saved_claim) { FactoryBot.create(:va526ez) }
+    let(:submitted_claim_id) { 600_130_094 }
```
Comment (freeheeling) [inline spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb:20]: I don't see where `submitted_claim_id` is used in this test.

Comment (sethdarragile6) [reply spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb]: copy/paste remnant. killed it.

File: spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  describe '.perform_async' do
+    let(:saved_claim) { FactoryBot.create(:va526ez) }
+    let(:submission) do
+      create(:form526_submission,
+             user_uuid: user.uuid,
+             auth_headers_json: auth_headers.to_json,
+             saved_claim_id: saved_claim.id)
+    end
```
Comment (freeheeling) [inline spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb:25]: For simplicity, I think this now can all be distilled down to 
```
let(:form526_submission) { create(:form526_submission) }
```
and then below:
```
 let(:form526_job_status) { create(:form526_job_status, :poll_form526_pdf, form526_submission:, job_id: 1) }
``` 
but the setup nevertheless works as is.

Comment (sethdarragile6) [reply spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb]: Quite right, thanks! Implemented 👍

File: spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+RSpec.describe Lighthouse::PollForm526Pdf, type: :job do
+  subject { described_class }
+
+  before do
+    Sidekiq::Job.clear_all
+    allow_any_instance_of(Auth::ClientCredentials::Service).to receive(:get_token).and_return('fake_access_token')
+  end
+
+  let(:user) { FactoryBot.create(:user, :loa3) }
```
Comment (freeheeling) [inline spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb:13]: Nice clean up. These aren't needed any longer either.

Comment (sethdarragile6) [reply spec/sidekiq/lighthouse/poll_form526_pdf_spec.rb]: Killed 'em! Thanks for your help on this, there were a lot more weeds to ~~poll~~ pull than I realized!

**Review by freeheeling — APPROVED** [review]
Thanks for addressing my concerns.

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**